src/ig/IGExporter.ts

```diff
@@ -1338,96 +1342,169 @@ export class IGExporter {
+
+    // Update IG.definition.parameter
+    this.ig.definition.parameter.forEach(parameter => {
+      const guideParameterCodes: string[] =
+        this.fhirDefs
+          .fishForFHIR('http://hl7.org/fhir/guide-parameter-code', Type.CodeSystem)
+          ?.concept.map((concept: CodeSystemConcept) => concept.code) ?? [];
+
+      const code = parameter.code as string;
+      parameter.code = { code };
+      const parsedCode = parseCodeLexeme(code);
+
+      if (parsedCode.code && parsedCode.system) {
+        // If we can parse the code and we have a system and a code provided, we should use that.
+        parameter.code.code = parsedCode.code;
+        parameter.code.system = parsedCode.system;
+      } else if (guideParameterCodes.some(c => c === code)) {
+        // Otherwise, only a code was provided, so we check if it is in the bound VS
+        parameter.code.system = 'http://hl7.org/fhir/guide-parameter-code';
+      } else {
+        // If the code is not in the VS in the R5 IG resource, we default the system
+        // based on https://chat.fhir.org/#narrow/stream/179252-IG-creation/topic/Unknown.20FHIRVersion.20code.20'5.2E0.2E0-ballot'/near/298697304
+        parameter.code.system = 'http://hl7.org/fhir/tools/CodeSystem/ig-parameters';
+      }
+    });
+
+    // Add new copyrightLabel property
+    if (this.config.copyrightLabel) {
+      this.ig.copyrightLabel = this.config.copyrightLabel;
+    }
 
-      // Add new copyrightLabel property
-      if (this.config.copyrightLabel) {
-        this.ig.copyrightLabel = this.config.copyrightLabel;
+    // Add new versionAlgorithm property
+    if (this.config.versionAlgorithmString) {
+      this.ig.versionAlgorithmString = this.config.versionAlgorithmString;
+    } else if (this.config.versionAlgorithmCoding) {
+      this.ig.versionAlgorithmCoding = this.config.versionAlgorithmCoding;
+    }
+
+    // Add new dependsOn.reason property
+    this.ig.dependsOn?.forEach(dependency => {
+      const configDependency = this.config.dependencies?.find(
+        d => d.packageId === dependency.packageId
+      );
+      if (configDependency.reason) {
+        dependency.reason = configDependency.reason;
       }
+    });
+  }
 
-      // Add new versionAlgorithm property
-      if (this.config.versionAlgorithmString) {
-        this.ig.versionAlgorithmString = this.config.versionAlgorithmString;
-      } else if (this.config.versionAlgorithmCoding) {
-        this.ig.versionAlgorithmCoding = this.config.versionAlgorithmCoding;
+  // If an R4 IG uses any new properties from R5, they should be included as extensions.
+  // Converting between FHIR versions is documented here: http://hl7.org/fhir/2022Sep/versions.html#extensions
+  addR5ExtensionsToR4() {
+    // Any new properties added to the IG resource in R5 should be included on extensions in R4 IGs
+    this.ig.definition.resource.forEach(resource => {
+      // Assign IG.definition.resource.profile to an extensions if provided
```

Comment:
This one is a little tricky because R4 has a _mostly_ equivalent property: `exampleCanonical` -- but it's `0..1` instead of `0..*`.  I think that maybe we should try to be smart about leveraging the existing R4 properties when it makes sense and only falling back to extensions when we need to represent something we can't say using R4 properties.  Maybe something like this?
* If `profile?.length === 1`, then `exampleCanonical = profile[0]`
* Else if `profile?.length > 1` , then `exampleCanonical = profile[0]` and add profile extension
* Else if `isExample != null` and `exampleCanonical == null`, then `exampleBoolean = isExample`

I think that basically says that R5 properties take precedence but we will represent them as R4 properties when possible.  It's a little tricky because in R5 `isExample` and `profile` can exist at the same time, but in R4, `example[x]` can only be `exampleCanonical` or `exampleBoolean` (and never both at the same time).  But I think maybe the pseudo-code above is right?  What do you think? Is this even a good idea?

Reply:
I think this is right. I think I missed some of the updates to existing R4 properties from the R5 properties potentially included in configuration, which is maybe also what happened with some of the `page` properties further down too.

I think the psuedo-code makes sense, with one question. If `profile?.length > 1`, should the extension that is added include all the profiles provided or just include `profile[1]` and up since `profile[0]` is already represented?

Reply:
I think it should include all the profiles. I'd rather it be a little redundant but complete. Otherwise if a tool sees the profile extension and _only_ looks at that, then it's missing something.  So I guess we're basically saying "Here is the _limited_ R4 representation (limited by cardinality in R4) and here is the _complete_ R5 representation (no cardinality limitation).  But if `profile` is only length 1, then R4 is not limited and therefore R5 is not needed.

Reply:
Updated in [32aa694](https://github.com/FHIR/sushi/pull/1189/commits/32aa694e112d618781fdf90c598633cbb24334d2).